src/api-binder.ts

```diff
@@ -467,14 +477,27 @@ export class APIBinder {
 			{
 				method: 'PATCH',
 				url: endpoint,
-				body: stateDiff,
+				body,
 			},
 			this.cachedBalenaApi.passthrough,
 		);
 
 		await this.cachedBalenaApi._request(requestParams);
 	}
 
+	// Returns an object that contains only status fields relevant for the local mode.
+	// It basically removes information about applications state.
+	public stripDeviceStateInLocalMode(
+		state: DeviceApplicationState,
+	): DeviceApplicationState {
+		const result = _.cloneDeep(_.pick(state, 'local'));
```

Comment:
I think this might well be better as
```suggestion
		const result = {
			local: _.cloneDeep(_.omit(state.local, 'apps', 'is_on__commit', 'logs_channel'));
		}
```
because it keeps the same behaviour without needing to do the `cloneDeep` on properties that are just going to be removed immediately after